vermouth/processors/water_bias.py

```diff
@@ -0,0 +1,151 @@
+# Copyright 2023 University of Groningen
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from .processor import Processor
+from ..graph_utils import make_residue_graph
+from ..rcsu.go_utils import get_go_type_from_attributes, _get_bead_size, _in_resid_region
+from ..gmx.topology import NonbondParam
+
+class ComputeWaterBias(Processor):
+    """
+    Processor which computes the water  bias for
+    the Martini Go and Martini IDP model.
+
+    The water bias strength is defined per secondary
+    structure element in `water_bias` and assigned if
+    `auto_bias` is set to True. Using the `idr_regions`
+    argument the water_bias can be changed for
+    intrinsically disordered regions (IDRs). The IDR
+    bias superseeds the auto bias.
+
+    This Processor updates the system.gmx_topology_params
+    attribute.
+
+    ** Subclassing **
+    If the procedure by which to assign the water bias is
+    to be changed this processor is best subclassed and the
+    assign_residue_water_bias method overwritten.
+    """
+
+    def __init__(self,
+                 auto_bias,
+                 water_bias,
+                 idr_regions):
+        """
+
+
+        Parameters
+        ----------
+        auto_bias: bool
+            apply the automatic secondary structure
+            dependent water biasing
+        water_bias: dict[str, float]
+            a dict of secondary structure codes and
+            epsilon value for the water bias in kJ/mol
+        idr_regions:
+            regions defining the IDRs
+        prefix: str
+            prefix of the Go virtual-site atomtypes
+        system: vermouth.system.System
+            the system of the molecules is used for
+            storing the nonbonded parameters
+        """
+        self.water_bias = water_bias
+        self.auto_bias = auto_bias
+        self.idr_regions = idr_regions
+        self.system = None
+
+    def assign_residue_water_bias(self, molecule, res_graph):
+        """
+        Assign the residue water bias for all residues
+        with a secondary structure element or that are
+        defined by the region selector. Region selectors
+        supercede the auto assignment.
+
+        Parameters
+        ----------
+        molecule: :class:vermouth.molecule.Molecule
+            the molecule
+        res_graph: :class:vermouth.molecule.Molecule
+            the residue graph of the molecule
```

Comment:
```suggestion
        molecule: :class:`vermouth.molecule.Molecule`
            the molecule
        res_graph: :class:`vermouth.molecule.Molecule`
            the residue graph of the molecule
```

It's either ``:class:`a.b.c` ``, or `a.b.c`